crates/ef-testing/src/sequencer/types.rs

```diff
@@ -0,0 +1,45 @@
+use reth_primitives::Address;
+use starknet::core::types::FieldElement;
+use starknet_api::{
+    core::{ContractAddress, PatriciaKey},
+    hash::StarkFelt,
+    StarknetApiError,
+};
+
+#[derive(Debug, Clone, Copy)]
+pub struct FeltSequencer(FieldElement);
+
+impl From<FieldElement> for FeltSequencer {
+    fn from(felt: FieldElement) -> Self {
+        Self(felt)
+    }
+}
+
+impl From<FeltSequencer> for FieldElement {
+    fn from(felt: FeltSequencer) -> Self {
+        felt.0
+    }
+}
+
+impl From<Address> for FeltSequencer {
+    fn from(address: Address) -> Self {
+        let address = FieldElement::from_byte_slice_be(&address.0[..]).unwrap(); // safe unwrap since Address is 20 bytes
+        Self(address)
+    }
+}
+
+impl From<FeltSequencer> for StarkFelt {
+    fn from(felt: FeltSequencer) -> Self {
+        StarkFelt::from(felt.0)
+    }
+}
+
+impl TryFrom<FeltSequencer> for ContractAddress {
```

Comment:
how could a Felt not fit into a contract address?

Reply:
Contract address is a wrapper around PatriciKey, and PatriciaKey upper bound is 2^251, lower than the felt limit